src/CSnakes.Runtime/Python/GIL.cs

```diff
@@ -62,7 +70,7 @@ public void Dispose()
             {
                 CPythonAPI.Py_DecRefRaw(handle);
             }
-            CPythonAPI.PyGILState_Release(gilState);
+            pythonThreadState = CPythonAPI.PyEval_SaveThread();

```

Comment:
The tracked thread state should be the returned one, right?  It may be worth verifying that's the case, e.g. with an assert.